### DISCUSSION TITLE
Should we make browser text selection configurable?

### DISCUSSION THREAD

**Discussion by icflorescu** [icflorescu/mantine-datatable#15] [Ideas]
For the moment, the browser text selection functionality is disabled by design (with a `user-select: none` CSS rule) when the `DataTable` is in [record selection mode](https://icflorescu.github.io/mantine-datatable/examples/records-selection).

I did this because I found the browser text selection to be confusing (especially on touch-screen devices) when working in record selection mode.

But, since [some see this as a bug](https://github.com/icflorescu/mantine-datatable/issues/14), maybe the intended behavior wasn't such a good idea.

Text selection is also disabled for the column headers at the moment.

So, let's consider the idea of allowing text selection by default and make it configurable.

**Please provide your feedback on the questions below:**

- How should we call the property? Would something like `disableTextSelection` be fine? 
- Should we create to properties to make the behavior separately configurable for column headers and rows? In this case, how do we call the properties? Would `disableColumnHeaderTextSelection` and `disableRowTextSelection` be fine? The names would be a mouthful, I know, but their purpose would be clear enough.

**Comment by Daxble** [comment]
I think having it be an option would be great, for apps that are super data heavy it's really a nice option to copy things directly from the table. Another solution I've used in the meantime, is adding click to copy on important columns.

**Reply by icflorescu** [reply]
I'm planning to add the options in a matter of days, anyway.

**Reply by Daxble** [reply]
> I'm planning to add the options in a matter of days, anyway.

Much appreciated!

**Comment by icflorescu** [comment]
Almost done.
I thought about it and I don't think it makes sense to have two distinct properties.
So, by default text selection is going to be enabled, we'll have a property called `textSelectionDisabled`.
I'll also update the docs accordingly.

**Reply by icflorescu** [reply]
Done.
